engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/AnyHashCodeNode.java

```diff
@@ -0,0 +1,78 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.*;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+
+import java.math.BigInteger;
+
+@BuiltinMethod(
+    type = "Any",
+    name = "hash_code",
+    description = "Gets the Hashcode value of an Enso object.")
+public abstract class AnyHashCodeNode extends Node {
+  private static final BigInteger bigInteger_MAX_LONG = BigInteger.valueOf(Long.MAX_VALUE);
+  private static final BigInteger bigInteger_MIN_LONG = BigInteger.valueOf(Long.MIN_VALUE);
+
+  public static AnyHashCodeNode build() {
+    return AnyHashCodeNodeGen.create();
+  }
+
+  public abstract Object execute(Object _this);
+
+  private static Long combineTypeAndHashCodes(int typeCode, int hashCode) {
+    return Long.valueOf(typeCode << 7 + typeCode + hashCode);
+  }
+
+  @Specialization
+  Object doBoolean(Boolean _this) {
+    // _this.getClass().toString().hashCode() == 1335156652
```

Comment:
I don't like putting this into comments and having baked-in constants.

Cannot we make these `static final` constants computed at initialization time?

i.e. somewhere below:
```
public static final long booleanClassHashcode = Boolean.class.toString().hashCode();
```